File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Luis/Gripper/Gripper.java
package org.firstinspires.ftc.teamcode.Luis.Gripper;

import com.qualcomm.robotcore.hardware.HardwareMap;
import com.seattlesolvers.solverslib.command.SubsystemBase;
import com.seattlesolvers.solverslib.hardware.ServoEx;
import com.seattlesolvers.solverslib.hardware.SimpleServo;

import org.firstinspires.ftc.robotcore.external.navigation.AngleUnit;

public class Gripper extends SubsystemBase {

    private final SimpleServo servo;
    private boolean isOpen = false;
    private HardwareMap hardwareMap;

    public Gripper(HardwareMap hardwareMap) {
        this.hardwareMap = hardwareMap;
        servo = new SimpleServo(hardwareMap, "gogo", 0.0, 180.0);
        close();
    }

    public void open() {
        if (!isOpen) {
            servo.turnToAngle(90.0);
            isOpen = true;
        }
    }

    public void close() {
        if (isOpen) {
            servo.turnToAngle(0.0);
            isOpen = false;
        }
    }

    public double getPosition() {
        return servo.getPosition();
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/arm/gripper/Gripper.kt
package org.firstinspires.ftc.teamcode.arm.gripper

import com.qualcomm.hardware.bosch.BNO055IMU
import com.qualcomm.robotcore.hardware.HardwareMap
import com.seattlesolvers.solverslib.command.SubsystemBase
import com.seattlesolvers.solverslib.hardware.ServoEx
import com.seattlesolvers.solverslib.hardware.SimpleServo
import org.firstinspires.ftc.robotcore.external.navigation.AngleUnit

class Gripper(private val config: GripperConfig, private val hardwareMap: HardwareMap): SubsystemBase() {

    private lateinit var servo: ServoEx

    private var isOpen: Boolean = false

    fun open() {
        servo.takeUnless { isOpen }?.position = 1.0
        isOpen = true
    }

    fun close() {
        servo.takeIf { isOpen }?.position = 0.0
        isOpen = false
    }

    fun getPosition() = servo.position

    init {
        require(config.maximumAngleRange > config.minimumAngleRange && config.minimumAngleRange > 0) {
            "The maximum movement range must be greater than the minimum one, this one must be also greater than zero"
        }
        configureServo()
    }

    private fun configureServo() {

        servo = SimpleServo(hardwareMap, config.servoId, config.minimumAngleRange, config.maximumAngleRange,
            AngleUnit.DEGREES)

        servo.inverted = config.isInverted

    }
}